candi.sh

```diff
@@ -605,6 +599,40 @@ guess_platform() {
     fi
 }
 
+guess_ostype() {
+    # Try to guess the operating system type (ostype)
+    if [ -f /usr/bin/cygwin1.dll ]; then
+        echo cygwin
+
+    elif [ -f /etc/fedora-release ]; then
+	    echo linux
+    
+    elif [ -f /etc/redhat-release ]; then
+        echo linux
+    
+    elif [ -x /usr/bin/sw_vers ]; then
+        echo macos
+    
+    elif [ -x /usr/bin/lsb_release ]; then
+        local DISTRO=$(lsb_release -i -s)
+        local CODENAME=$(lsb_release -c -s)
+        local DESCRIPTION=$(lsb_release -d -s)
+        case ${DISTRO}:${CODENAME}:${DESCRIPTION} in
+            *:*:*Ubuntu*\ 12*)     echo linux;;
```

Comment:
Why are you adding this detection using lsb_release? It is not guaranteed to be available, see https://www.unixmen.com/linux-troubleshooting-fix-lsb_release-command-found-centos/ for example